chains/tendermint/prover.go

```diff
@@ -161,50 +129,72 @@ func (pr *Prover) SetupHeader(dstChain core.LightClientIBCQueryierI, srcHeader c
 
 	// inject TrustedValidators into header
 	h.TrustedValidators = valSet
-	return &h, nil
+	return []core.Header{&h}, nil
 }
 
-func lightError(err error) error { return fmt.Errorf("light client: %w", err) }
+// GetLatestFinalizedHeader returns the latest finalized header
+func (pr *Prover) GetLatestFinalizedHeader() (latestFinalizedHeader core.Header, err error) {
+	h, err := pr.UpdateLightClient()
+	if err != nil {
+		return nil, err
+	}
+	return h, nil
+}
 
-// UpdateLightWithHeader calls client.Update and then .
-func (pr *Prover) UpdateLightWithHeader() (header core.HeaderI, provableHeight int64, queryableHeight int64, err error) {
+/* Local LightClient implementation */
+
+// GetLatestLightHeight uses the CLI utilities to pull the latest height from a given chain
+func (pr *Prover) GetLatestLightHeight() (int64, error) {
+	db, df, err := pr.NewLightDB()
+	if err != nil {
+		return -1, err
+	}
+	defer df()
+
+	client, err := pr.LightClient(db)
+	if err != nil {
+		return -1, err
+	}
+
+	return client.LastTrustedHeight()
+}
+
+func (pr *Prover) UpdateLightClient() (core.Header, error) {
```

Comment:
You can unexport this function.

Reply:
https://github.com/hyperledger-labs/yui-relayer/blob/07ad300cb601f5d4268ef3c11061cdb0f53e8191/chains/tendermint/cmd/light.go#L106
It is also referenced in the option of the `update` command.